test/oracles/adapters/balancer/BalancerV2PriceOracle.test.ts

```diff
@@ -0,0 +1,225 @@
+const USDC_WFTM_BTC_ETH_POOL = "0xf3A602d30dcB723A74a0198313a7551FEacA7DAc";
+
+describe("Balancer V2 Price Oracle", function () {
+    let addressRegistry: IExternalContractsRegistry;
+    let owner: SignerWithAddress, user: SignerWithAddress, architect: SignerWithAddress;
+    let oracle: BalancerV2PriceOracle;
+    let roles: ExodiaRoles;
+
+    before(async function () {
+        [owner, user, architect] = await xhre.ethers.getSigners();
+        addressRegistry = externalAddressRegistry.forNetwork(await getNetwork());
+    });
+
+    beforeEach(async function () {
+        await deployments.fixture([EXODIA_ROLES_DID, BALANCER_V2_PRICE_ORACLE_DID]);
+        const oracleDeployment = await get<BalancerV2PriceOracle__factory>(
+            "BalancerV2PriceOracle"
+        );
+        oracle = await oracleDeployment.contract;
+        const rolesDeployment = await get<ExodiaRoles__factory>("ExodiaRoles");
+        roles = await rolesDeployment.contract;
+
+        await roles.addArchitect(architect.address);
+    });
+
+    it("Can't initialize with zero addresses", async function () {
+        await expect(
+            oracle.initialize(
+                ZERO_ADDRESS,
+                addressRegistry.BEETHOVEN_VAULT,
+                MINIMUM_UPDATE_INTERVAL
+            )
+        ).to.revertedWith("roles cannot be null address");
+        await expect(
+            oracle.initialize(roles.address, ZERO_ADDRESS, MINIMUM_UPDATE_INTERVAL)
+        ).to.revertedWith("vault cannot be null address");
+    });
+
+    it("Should be able to initialize", async function () {
+        await oracle.initialize(
+            roles.address,
+            addressRegistry.BEETHOVEN_VAULT,
+            MINIMUM_UPDATE_INTERVAL
+        );
+
+        expect(await oracle.vault()).to.equal(addressRegistry.BEETHOVEN_VAULT);
+    });
+
+    it("Can't initialize twice", async function () {
+        await oracle.initialize(
+            roles.address,
+            addressRegistry.BEETHOVEN_VAULT,
+            MINIMUM_UPDATE_INTERVAL
+        );
+
+        await expect(
+            oracle.initialize(
+                ZERO_ADDRESS,
+                addressRegistry.BEETHOVEN_VAULT,
+                MINIMUM_UPDATE_INTERVAL
+            )
+        ).to.revertedWith("Initializable: contract is already initialized");
+    });
+
+    describe("After initialization", function () {
+        beforeEach(async function () {
+            await oracle.initialize(
+                roles.address,
+                addressRegistry.BEETHOVEN_VAULT,
+                MINIMUM_UPDATE_INTERVAL
+            );
+        });
+
+        it("Only architect can set token oracle settings", async function () {
+            await expect(
+                oracle
+                    .connect(user)
+                    .setTokenOracle(
+                        addressRegistry.BEETS,
+                        addressRegistry.FIDELIO_DUETTO,
+                        addressRegistry.FTM_USD_FEED
```

Comment:
Oracle for denominated asset should 0 to be more explicit and show we are not using for assets that are already denominated with FTM